### DISCUSSION TITLE
Custom GitHub Enterprise host name

### DISCUSSION THREAD

**Discussion by buhruf** [BetaHuhn/repo-file-sync-action#8] [Ideas] [answered]
Hi there,

You've done some incredible work here and my hat goes off to you! Bravo!

I just had a comment about the following line of code in `src/git.js`: 

```
const gitUrl = `https://${ GITHUB_TOKEN }@github.com/${ repo.fullName }.git`
```

It looks like this will only work for standard GitHub repositories, but not for those hosted on enterprise servers with custom host names (example: `github.custom.host`). 

During the `Repo Sync Action`,  the action will resolve the path as:

```
Repository Info
Https Url	: https://github.com/exampleUser/exampleRepo
```

Which will result in a fatal error:

```
fatal: could not read Password for 'https://***@github.com': No such device or address
```

And as far as I can tell, a host parameter currently can't be prepended in the sync.yml file required for the action to run. This is, AFAIK, due to how the host is constructed in the URL in the git.js script.

Have  you thought about adding another property to the sync.yml file where users could enter the host on which the repository is stored? 

Just a thought! At any rate, you're doing fantastic work! 

Best regards,

D

**Comment by BetaHuhn** [comment, marked as answer]
Hey 👋 

First of all, thanks for your kind words and your feedback!

> It looks like this will only work for standard GitHub repositories, but not for those hosted on enterprise servers with custom host names (example: github.custom.host).

You are totally correct, the Action currently only works for repositories hosted on github.com. There's no real reason for this other than I haven't thought about enterprise repositories 😄 

> Have you thought about adding another property to the sync.yml file where users could enter the host on which the repository is stored?

We could allow an entire URL with the host as the key in the sync.yml file.

So instead of this:

```yml
user/repo:
  - path/to/file.txt
```

You could do:

```yml
https://host/user/repo:
  - path/to/file.txt
```

Would this work for you @buhruf?

**Reply by buhruf** [reply]
Hi there,

Yes! I believe that would work quite nicely, @BetaHuhn!

**Reply by BetaHuhn** [reply]
Okay, I implemented this on the [`feature/host`](https://github.com/BetaHuhn/repo-file-sync-action/tree/feature/host) branch.

You can give it a try by specifying the branch when using the Action:

```yml
- uses: BetaHuhn/repo-file-sync-action@feature/host
```

and then specifying your custom host like this (more info in the [README](https://github.com/BetaHuhn/repo-file-sync-action/tree/feature/host#custom-github-enterprise-host)):

```yml
https://custom.host/user/repo:
  - path/to/file.txt
```

> **Note:** The key has to start with `http` to indicate that you want to use a custom host.

@buhruf please let me know if everything works with a custom host, as I can't test this myself.

Once you confirm that it's working I will merge this into master.

**Reply by buhruf** [reply]
@BetaHuhn, well done! I can confirm that the changes on the `feature/host` branch work as intended. 

I believe you can safely merge this change 😄 . 

Thanks again!

**Reply by BetaHuhn** [reply]
Done!

This is now a part of [v1.2.0](https://github.com/BetaHuhn/repo-file-sync-action/releases/tag/v1.2.0) 🎉

**Reply by buhruf** [reply]
🍾 🎉 💯
